src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Parser.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Generators
+{
+    public partial class EventSourceGenerator
+    {
+        private class Parser
+        {
+            private readonly CancellationToken _cancellationToken;
+            private readonly Compilation _compilation;
+            private readonly Action<Diagnostic> _reportDiagnostic;
+
+            public Parser(Compilation compilation, Action<Diagnostic> reportDiagnostic, CancellationToken cancellationToken)
+            {
+                _compilation = compilation;
+                _cancellationToken = cancellationToken;
+                _reportDiagnostic = reportDiagnostic;
+            }
+
+            public EventSourceClass[] GetEventSourceClasses(List<ClassDeclarationSyntax> classDeclarations)
+            {
+                var autogenerateAttribute = _compilation.GetTypeByMetadataName("System.Diagnostics.Tracing.EventSourceAutoGenerateAttribute");
```

Comment:
var => whatever it is

Same for all vars :)

Reply:
Done